test/ireduce.cpp

```diff
@@ -185,3 +194,146 @@ TEST(IndexedReduce, MaxReduceDimensionHasSingleValue)
     ASSERT_TRUE(allTrue<bool>(mm == data));
     ASSERT_TRUE(allTrue<bool>(indx == 0));
 }
+
+TEST(IndexedReduce, MinNaN)
+{
+    float test_data[] = { 1, NAN, 5, 0.1, NAN, -0.5, NAN, 0 };
+    int rows = 4;
+    int cols = 2;
+    array a(rows, cols, test_data);
+
+    float gold_min_val[] = { 0.1, -0.5 };
+    int gold_min_idx[] = { 3, 1 };
+
+    array min_val;
+    array min_idx;
+    min(min_val, min_idx, a);
+
+    vector<float> h_min_val(cols);
+    min_val.host(&h_min_val[0]);
+
+    vector<int> h_min_idx(cols);
+    min_idx.host(&h_min_idx[0]);
+
+    for (int i = 0; i < cols; i++) {
+        ASSERT_FLOAT_EQ(h_min_val[i], gold_min_val[i]);
+    }
+
+    for (int i = 0; i < cols; i++) {
+        ASSERT_EQ(h_min_idx[i], gold_min_idx[i]);
+    }
+}
+
+TEST(IndexedReduce, MaxNaN) 
+{ 
+    float test_data[] = { 1, NAN, 5, 0.1, NAN, -0.5, NAN, 0 }; 
+    int rows = 4; 
+    int cols = 2; 
+    array a(rows, cols, test_data); 
+ 
+    float gold_max_val[] = { 5.0, 0.0 }; 
```

Comment:
add f to the end of those constants to avoid narrowing warnings. i.e. `{5.0f, 0.f};`

Reply:
Seems like there are other places where this could be applied.

Reply:
👍